baseapp/util_test.go

```diff
@@ -1,68 +1,153 @@
-package baseapp
+package baseapp_test
 
 import (
-	"github.com/cosmos/cosmos-sdk/snapshots"
-	"github.com/cosmos/cosmos-sdk/types"
-)
+	"context"
+	"encoding/json"
+	"testing"
 
-// TODO: Can be removed once we move all middleware tests into x/auth/middleware
-// ref: #https://github.com/cosmos/cosmos-sdk/issues/10282
+	"github.com/stretchr/testify/require"
+	tmtypes "github.com/tendermint/tendermint/types"
 
-// CheckState is an exported method to be able to access baseapp's
-// checkState in tests.
-//
-// This method is only accessible in baseapp tests.
-func (app *BaseApp) CheckState() *state {
-	return app.checkState
-}
+	runtimev1alpha1 "cosmossdk.io/api/cosmos/app/runtime/v1alpha1"
+	appv1alpha1 "cosmossdk.io/api/cosmos/app/v1alpha1"
+	authmodulev1 "cosmossdk.io/api/cosmos/auth/module/v1"
+	bankmodulev1 "cosmossdk.io/api/cosmos/bank/module/v1"
+	mintmodulev1 "cosmossdk.io/api/cosmos/mint/module/v1"
+	paramsmodulev1 "cosmossdk.io/api/cosmos/params/module/v1"
+	stakingmodulev1 "cosmossdk.io/api/cosmos/staking/module/v1"
+	txmodulev1 "cosmossdk.io/api/cosmos/tx/module/v1"
+	"cosmossdk.io/core/appconfig"
+	"cosmossdk.io/depinject"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	"github.com/cosmos/cosmos-sdk/testutil/mock"
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/cosmos/cosmos-sdk/x/auth"
+	_ "github.com/cosmos/cosmos-sdk/x/auth/tx/module"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	_ "github.com/cosmos/cosmos-sdk/x/bank"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	_ "github.com/cosmos/cosmos-sdk/x/mint"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	_ "github.com/cosmos/cosmos-sdk/x/params"
+	_ "github.com/cosmos/cosmos-sdk/x/staking"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
 
-// DeliverState is an exported method to be able to access baseapp's
-// deliverState in tests.
-//
-// This method is only accessible in baseapp tests.
-func (app *BaseApp) DeliverState() *state {
-	return app.deliverState
-}
+// GenesisStateWithSingleValidator initializes GenesisState with a single validator and genesis accounts
+// that also act as delegators.
+func GenesisStateWithSingleValidator(t *testing.T, codec codec.Codec, builder *runtime.AppBuilder) map[string]json.RawMessage {
```

Comment:
Maybe this is what you meant in your comment, but shouldn't this be in `/testutil/sims` instead?

Reply:
More or less; that function's signature looks like

```
func GenesisStateWithSingleValidator(t *testing.T, app *SimApp) GenesisState
```

so I can't use it here.  Maybe we can live with a bit of code duplication until it's more clear?